hooks/save-data/index.js

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useState } from 'react';
+
+const UNSUPPORT_MESSAGE = 'The Save Data API is not supported on this platform.';
+const SAVE_DATA_MODE = {
+  ON: 'on',
+  OFF: 'off'
+};
+
+const useSaveData = () => {
+  let initialSaveData;
+  if ('connection' in navigator) {
+    initialSaveData = navigator.connection.saveData ? SAVE_DATA_MODE.ON : SAVE_DATA_MODE.OFF;
+  } else {
+    initialSaveData = {unsupportMessage: UNSUPPORT_MESSAGE};
+  }
+
+  const [saveData, setSaveData] = useState(initialSaveData);
+
+  return {saveData, setSaveData};
+};
+
+export {
+  useSaveData,
+  // exported for Unit Tests
+  UNSUPPORT_MESSAGE,
+  SAVE_DATA_MODE
+};
```

Comment:
Similar to the comments on the other thread, I think we should refactor this to only pass along the minimal useful information and move unit-test specific values to test configuration.

Reply:
Yes, of course. It's in progress in the other branch. No worries. :)